lib/posix/spawn/child.rb

```diff
@@ -262,12 +288,10 @@ def read_and_write(input, stdin, stdout, stderr, timeout=nil, max=nil)
           end
 
           # maybe we've hit our max output
-          if max && ready[0].any? && (@out.size + @err.size) > max
+          if max && ready[0].any? && (@stdout_buffer.size + @stderr_buffer.size) > max
```

Comment:
again, we can't assume there is a `#size` method on these objects...

I think we should probably keep a local count of the bytes we have written instead of calling `#size` anyway. We can't trust these objects any more as they could be anything.